src/main/java/com/io/image/manager/controller/ImageController.java

```diff
@@ -58,6 +67,13 @@ public ImageController(ImageService imageService, PrometheusMeterRegistry mr, Ap
             }
             writer = new BufferedWriter(new FileWriter(logDir + "/IM_log.txt", true));
         }
+
+        connectionManager = new PoolingHttpClientConnectionManager();
```

Comment:
```suggestion
        var poolManager = new PoolingHttpClientConnectionManager();
```
And then you can assign
```java
this.connectionManager = poolManager
```
to avoid so many casts.

Reply:
And I would opt for making the http client a service instead of another variable inside image controller. Then you wouldn't have to pass it as a param everywhere.

Reply:
I don't necessarily see how turning it into service would simplify the code, it adds a whole other class just to remove passing another argument in two method calls

Reply:
And with dependency injection it clears a lot. You don't depend on the argument and have the whole class injected that will do the job.